packages/program/src/contract-call-script.ts

```diff
@@ -97,6 +97,10 @@ const getSingleCallInstructions = (
 };
 // Given a list of contract calls, create the actual opcodes used to call the contract
 function getInstructions(offsets: CallOpcodeParamsOffset[], outputs: CallOutputInfo[]): Uint8Array {
+  if (!offsets.length) {
+    return new Uint8Array();
```

Comment:
Unfortunately, as I said in the other comment, I can't see how we'd successfully circumvent the problem without the complete WASM initialization.

Reply:
Couldn't this await be done at the earliest `async` point in the call stack?

For example, here:

https://github.com/FuelLabs/fuels-ts/blob/66782ed56cff4cee78320b0d90155e6671c56fd5/packages/program/src/functions/base-invocation-scope.ts#L160-L162

It'd look something like this:

```ts
protected async prepareTransaction() {
  await asm.initWasm();

  // Update request scripts before call 
  this.updateScriptRequest(); 
```

Reply:
My bad. It was too good to be true.

However, I'm afraid that simply calling the `initWasm()` at the top of `instruction-set.ts` will not be enough, as we're not `await`-ing for it. The problem is not calling `initWasm()` per se since it is automatically called internally at startup. The problem is that consumers must `await` for it.

Jest is giving us **_false positives_** due to its delayed initialization. The interval between the time the `import`s are processed at the top of the test files and the first test being run seems enough for the WASM to initialize.

We need to use **_dynamic_** imports in the tests to work around this and eliminate this delay.

The example below will fail until you uncomment the `await asm.initWasm()`.


```ts
import { InstructionSet } from './instruction-set';

describe('Instruction Set', () => {
  test('Can create instruction set immediately', async () => {
    const asm = await import('@fuels/vm-asm');

    // // @ts-expect-error DTS file is missing `initWasm` declaration
    // await asm.initWasm();

    const instHex = new InstructionSet(
      asm.movi(0x10, 11),
      asm.movi(0x11, 22),
      asm.lw(0x11, 0x11, 0),
      asm.movi(0x12, 33),
      asm.lw(0x12, 0x12, 0),
      asm.movi(0x13, 44),
      asm.call(0x10, 0x12, 0x13, 0x11)
    ).toHex();

    expect(instHex).toEqual('0x7240000b724400165d451000724800215d492000724c002c2d4124d1');
  });
});
```

> [!WARNING]
> Remember that after the first `await asm.initWasm()` is processed, all other tests should pass even without the `await`. This is very sensitive and can result in flakiness if forgotten.